resotocore/resotocore/infra_apps/package_manager.py

```diff
@@ -0,0 +1,264 @@
+from typing import AsyncGenerator, Optional, Union, List, Dict
+import asyncio
+from asyncio import Lock
+import subprocess
+from pathlib import Path
+import aiofiles
+from aiofiles import os as aos
+from attrs import frozen
+import aiohttp
+import shutil
+import hashlib
+from collections import defaultdict
+
+from resotocore.infra_apps.manifest import AppManifest
+from resotocore.db.async_arangodb import AsyncArangoDB
+from resotocore.db.entitydb import EntityDb, ArangoEntityDb
+from resotocore.config import ConfigHandler
+from resotocore.config import ConfigEntity
+from resotocore.ids import InfraAppName, ConfigId
+from resotocore.model.model import Kind, ComplexKind
+from resotocore.model.typed_model import from_js
+from logging import getLogger
+
+logger = getLogger(__name__)
+
+
+@frozen
+class FromHttp:
+    http_url: str
+
+
+@frozen
+class FromGit:
+    git_url: str
+
+
+InstallationSource = Union[FromHttp, FromGit]
+
+
+@frozen
+class InfraAppPackage:
+    manifest: AppManifest
+    source: InstallationSource
+
+
+PackageEntityDb = EntityDb[InfraAppName, InfraAppPackage]
+
+
+def app_manifest_entity_db(db: AsyncArangoDB, collection: str) -> ArangoEntityDb[InfraAppName, InfraAppPackage]:
+    return ArangoEntityDb(db, collection, InfraAppPackage, lambda k: InfraAppName(k.manifest.name))
```

Comment:
only used in tests?
I tried to bundle all db access functionality in `DbAccess` - consider moving the functionality there.

Reply:
At the moment - yes. I still need to glue things together, and at that point, it will be used in production.